kafka-message-signing/src/main/java/com/alliander/osgp/kafka/message/signing/MessageSigner.java

```diff
@@ -192,13 +204,64 @@ public boolean verify(final SignableMessageWrapper<?> message) {
     }
   }
 
+  /**
+   * Verifies the signature of the provided {@code message}.
+   *
+   * @param message the message
+   * @param producerRecord the record to be verified
+   * @return {@code true} if the signature of the given {@code producerRecord} was verified; {@code false}
+   *     if not.
+   * @throws IllegalStateException if this message signer has a private key needed for signing
+   *     messages, but does not have the public key for signature verification.
+   * @throws UncheckedIOException if determining the bytes for the message throws an IOException.
+   * @throws UncheckedSecurityException if the signature verification process throws a
+   *     SignatureException.
+   */
+  public <T> boolean verifyRecordHeader(final SignableMessageWrapper<T> message, final ProducerRecord<String, T> producerRecord) {
```

Comment:
Als we de record header checken hoeft het niet perse gewrapt te worden dus mag ook gewoon `T` zijn in plaats van `SignableMessageWrapper<T>`